Modules/_ssl.c

```diff
@@ -2834,6 +2841,50 @@ _ssl__SSLSocket_verify_client_post_handshake_impl(PySSLSocket *self)
 #endif
 }
 
+#if HAVE_EXPORT_KEYING_MATERIAL
+/*[clinic input]
+_ssl._SSLSocket.export_keying_material
+    label: Py_buffer(accept={buffer, str, NoneType})
+    key_len: int
+    context: Py_buffer(accept={buffer, str, NoneType})
+[clinic start generated code]*/
+
+static PyObject *
+_ssl__SSLSocket_export_keying_material_impl(PySSLSocket *self,
+                                            Py_buffer *label, int key_len,
+                                            Py_buffer *context)
+/*[clinic end generated code: output=a9308312295b26cc input=bdcd503c149308b0]*/
+{
+    PyObject *out = NULL;
+    unsigned char *key;
+
+    if (key_len < 1) {
+        PyErr_SetString(PyExc_ValueError, "key_len must be positive");
+        return NULL;
+    }
+
+    out = PyBytes_FromStringAndSize(NULL, key_len);
+    if (out == NULL)
+        goto error;
+
+    key = (unsigned char *)PyBytes_AS_STRING(out);
+
+    if (SSL_export_keying_material(self->ssl,
+                                   key, key_len,
+                                   label->buf, label->len,
+                                   context->buf, context->len,
+                                   context->buf != NULL) != 1) {
```

Comment:
Please follow the error checking style of the module:

```suggestion
    if (!SSL_export_keying_material(self->ssl,
                                    key, key_len,
                                    label->buf, label->len,
                                    context->buf, context->len,
                                    context->buf != NULL)) {
```